### PR TITLE
added mongock scripts to update current measures

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-5586](https://jira.cms.gov/browse/MAT-5586)
(Optional) Related Tickets:

### Summary
added mongock scripts to update current measures

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/config/AddMeasureSetChangeUnit.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import io.mongock.api.annotations.ChangeUnit;
+import io.mongock.api.annotations.Execution;
+import io.mongock.api.annotations.RollbackExecution;
+import java.io.IOException;
+import java.util.List;
+
+@ChangeUnit(id = "add_measure_set", order = "1", author = "madie_dev")
+public class AddMeasureSetChangeUnit {
+
+  private final ObjectMapper objectMapper;
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/config/AddMeasureSetChangeUnit.java]: Doesn’t look like this is required

File: src/main/java/cms/gov/madie/measure/config/AddMeasureSetChangeUnit.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  public AddMeasureSetChangeUnit(ObjectMapper objectMapper) {
+    this.objectMapper = objectMapper;
+  }
+
+  @Execution
+  public void addMeasureSetValues(
+      MeasureSetRepository measureSetRepository, MeasureRepository measureRepository)
+      throws IOException {
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/config/AddMeasureSetChangeUnit.java]: No file operations here. This exception seems to be unnecessary

File: src/main/java/cms/gov/madie/measure/repositories/MeasureRepository.java [hunk 3]
```diff
@@ -24,6 +25,15 @@
...
       String user, Boolean active, String shared, Pageable page);
 
+  @Aggregation(
+      pipeline = {
+        "{'$sort': {'createdAt':1}}",
+        "{'$group': {'_id': '$_id',"
+            + "'measureSetId': {'$first':'$measureSetId'},"
+            + "'createdBy': {'$first':'$createdBy'}}}"
+      })
+  List<Measure> findDistinctByMeasureSetField();
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/repositories/MeasureRepository.java]: maybe `findOldestInMeasureSet` or something similar ?

File: src/main/java/cms/gov/madie/measure/config/AddMeasureSetChangeUnit.java (hunk 1 above)
Comment (sb-prateekkeerthi) [reply src/main/java/cms/gov/madie/measure/config/AddMeasureSetChangeUnit.java]: Thanks, I just pushed the changes

File: src/main/java/cms/gov/madie/measure/config/AddMeasureSetChangeUnit.java (hunk 2 above)
Comment (sb-prateekkeerthi) [reply src/main/java/cms/gov/madie/measure/config/AddMeasureSetChangeUnit.java]: Thanks, I just pushed the changes

File: src/main/java/cms/gov/madie/measure/repositories/MeasureRepository.java (hunk 3 above)
Comment (sb-prateekkeerthi) [reply src/main/java/cms/gov/madie/measure/repositories/MeasureRepository.java]: Done

**Review by adongare — APPROVED** [review]

**Pull request merged**